OJ-126

wenlinwwww opened the pull request:
## What has been done
- Changed UI for order history 
- Added pop-up window for each history
- Added filter for search history
- Added changed order status method

## What has been tested
### 1. For the user, clicking the avatar will show order history, but for the admin, it will show order management
![f8b4c4be459ee29f349dad910cc3ec2](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/d993ae6a-1619-43c9-b453-978c02d9049e)
![a7bcc0f41da8435a0e44a2e6cfdd9db](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/f4f84d6e-f1d5-4a91-a9fe-879346f33f02)

### 2. New UI style for user
![62543d6473f32c78c6228aff05fd005](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/55419776-8cdc-499a-ba50-26c8511876c5)
### When clicking the order, it shows the pop-up dialog:
![57bf4bcc18cefa7bc9b1880462aef4f](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/fdbbbd53-48cf-4d5f-bc0a-64a24600aef9)
### Sorted by order number:
![336ca6bb97a4d3028ad91ae022f723d](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/1e8c0468-b992-48e4-971f-9950134b085b)
### Sorted by price from low to high:
![4ae0bd1d1d462d5f7b98cda216fade8](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/ef5ddc72-7568-4d91-9bc8-cc26e9c1bd1b)
### Only show delivery and dine-in orders:
![fc07883485ac6aa191b5332cddac06e](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/510ac4c6-9488-42f7-b22d-8abb954dcc85)
### Only show orders with pending status:
![a31f0471db75979d040d8c11765ab15](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/7761a073-4233-447b-964a-f710260b1220)
### text search: if the user types some number, it will match with order number and total price.
![6847b3407a492d74573a853725580d1](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/6cb22675-bfe6-4bc4-bd52-6b3bdbe67c59)
![fc19e6a03081a6059f79caa731d2d4d](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/ae696570-d01e-4106-b943-be2663f0e851)

### If the user types # in the front, it will only search the orders with the order number.
![acc30bd9e662cbac5ae8fe2448414b4](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/9e598f98-ca19-4ed7-a6d7-dd81009d8e7d)
![a52809c47211098cf34f206ef3a8093](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/a9dccec7-08eb-426c-95c2-8620cf4ad5a8)

### If typing words, it will search the orders with that username.
![0c865e4f42c048893bc450ec4a349ec](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/ed81acf0-b793-4ef8-b1d9-bbda1a6444b6)

### 2. To change order status, click `modify order`:
![97058c8f9c14c95f1c9ed78feff817e](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/531f8c39-9418-4a5d-aa36-0ec270b3df20)
![fe0824ba90dbde88a2de3d37a25849f](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/21addd9f-83c5-4d11-80ac-2c1c048a30df)
![45180c8a3c1abc766915c73feae3598](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/885cc8e5-64bb-4055-b8ad-920bf749dfd9)
### after clicking the save button, a pop-up will close, but reopen it, it will show the updated status:
![6fc4902a96a932ca743b8426cd7c7de](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/24598b67-e9c4-42fc-a20e-bce1a6c3f683)
### If the admin clicks `reject order`, it will be deleted directly. (The original total order was 11, but after deleting the last one, only 10 left)
![cdb4f1216a3555325ae4d33e1ad5b60](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/bf6e18c5-82a2-45ee-953f-88a9afc2e824)
### 3. Pagination: each page displays up to 10 orders. If there are more than 10 orders, the additional orders are distributed to subsequent pages accordingly.
![086e713c8693087713b6e72494b350e](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/81b372e6-d075-4b3b-9e97-721f24023730)
![c1c28b62862fb4d5fddd10e74887d2a](https://github.com/ssxxxxgrace/OrderHere-frontend/assets/104260056/46f5b693-e069-47fe-97b4-37b9e6b1dc33)

wenlinwwww committed:
fixed place order issue & updated order management

SteveGuoh reviewed (commented):
@wenlinwwww  filter "Cancelled" and "In Transit" not working on my laptop.

It will basically not filter all orders with type "Cancelled" or "In Transit" regardless the box has been ticked or not.

<img width="483" alt="image" src="https://github.com/ssxxxxgrace/OrderHere-frontend/assets/127906105/05f64027-27a0-46ef-8d9e-188fc9f4da2b">

<img width="1235" alt="image" src="https://github.com/ssxxxxgrace/OrderHere-frontend/assets/127906105/ada4b98a-ea68-482a-9cbb-aab896514741">

<img width="1237" alt="image" src="https://github.com/ssxxxxgrace/OrderHere-frontend/assets/127906105/421c202c-0b68-4447-af8b-a44c897b5018">

wenlinwwww committed:
fixed typing error

wenlinwwww commented:
@SteveGuoh I have corrected some typing errors: 
changed `canceled` to `cancelled` and `in transite` to `in transit`. 
Now, the filter should work correctly for cancelled and in_transit statuses.
